content/en/blog/2024/otel-errors/index.md

```diff
@@ -0,0 +1,286 @@
+---
+title: Dude, Where's My Error? How OpenTelemetry Records Errors
+linkTitle: OTel Errors
+date: 2024-03-06
+author: >- # If you have only one author, then add the single name on this line in quotes.
+  [Reese Lee](https://github.com/reese-lee) (New Relic),
+  [Adriana Villela](https://github.com/avillela) (ServiceNow)
+draft: true # TODO: remove this line once your post is ready to be published
+# canonical_url: http://somewhere.else/ # TODO: if this blog post has been posted somewhere else already, uncomment & provide the canonical URL here.
+---
+Depending on the language(s) you’re used to developing in, you may have certain 
+ideas about what an error is, as well as what constitutes an exception and how 
+it should be handled. For example, Go does not have exceptions, partly to 
+discourage programmers from labeling too many ordinary errors as exceptional. 
+On the other hand, languages such as Java and Python provide built-in support 
+for throwing and catching exceptions. 
+
+When you begin at a place where different languages disagree about what an error 
+or exception is and how to handle them, where do you go when you need standardized 
+telemetry and error reporting across microservices written in those languages? 
+OpenTelemetry is the key to this answer, and is the lens through which we’ll 
+address the following, and more:
+
+* How an error is visualized in a backend may not be where you think it’ll be, 
+or how you expect it to look. 
+* How span kind affects error reporting. 
+* Errors reported by spans vs logs.
+
+## Errors vs Exceptions
+Before we get into how OTel deals with errors and exceptions, let’s establish 
+what they are, and how they’re different from each other. While there are 
+variations on the definitions of these terms, we’ve landed on the ones below, 
+which we’ll be using in this article:
+
+An **error** is an unexpected issue in a program that hinders its execution. 
+Examples include syntax errors, such as a missing semicolon or incorrect 
+indentation, and runtime errors, resulting from errors in logic.
+
+An **exception** is a type of runtime error that disrupts the normal flow of a 
+program. Examples include dividing by zero or accessing an invalid memory address. 
```

Comment:
Do we have some definitions in the spec on what an error, exception is, are your definitions similar to them? We should make sure that those things are consistent.

Reply:
Hm, so I actually looked for these, and there is no explicit definition in the spec that I was able to find. There are instead guidelines on how to handle errors and exceptions, perhaps I can write up a couple sentences to explain this -- would that work?

Reply:
What is important to me is that it is 100% clear to a reader that does definitions are not opentelemetry language but what you use in this article. So the following needs to be stronger:

> we’ve landed on the following ones, which we’ll be using in this article: